### PR TITLE
Added client invoices dashboard

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
### What
- Users with role client can log in and view invoices
- They can search for invoices. 
- They can view invoices of all statuses except `draft` and `declined`
- They can click on any invoice, view it, and make a payment.

### Preview
https://youtu.be/0c9gLloEycQ

### CODE REVIEW

File: app/javascript/src/components/ClientInvoices/Details/InvoiceInfo copy.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+
+import dayjs from "dayjs";
```
Comment (gowsik-ragunath) [inline app/javascript/src/components/ClientInvoices/Details/InvoiceInfo copy.tsx:3]: Please remove this file if not needed

File: app/javascript/src/components/ClientInvoices/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    const timer = setTimeout(() => {
+      setIsShrinkingSearchBox(false);
+      clearTimeout(timer);
+    }, 300);
+  };
+
+  return (
+    <div className="relative mt-6 mb-3 flex flex-wrap items-center justify-between md:justify-start lg:justify-between">
+      {isDesktop && <h2 className="header__title">Invoices</h2>}
+      {isDesktop ? (
```
Comment (gowsik-ragunath) [inline app/javascript/src/components/ClientInvoices/Header.tsx:95]:
```suggestion
      {isDesktop ? (
       <h2 className="header__title">Invoices</h2>
```

Move the header inside the ternary condition true block.

File: app/javascript/src/constants/index.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export enum Paths {
   LOGIN = "/login",
   SIGN_IN = "/user/sign_in",
   CLIENTS = "/clients",
-  INVOICES = "/invoices",
+  INVOICES = "/invoices/*",
```
Comment (gowsik-ragunath) [inline app/javascript/src/constants/index.tsx:39]: do we need *  at the end?

**Review by gowsik-ragunath — CHANGES_REQUESTED** [review]
Added a few comments please check

File: app/javascript/src/constants/index.tsx (hunk 3 above)
Comment (prasanthchaduvula) [reply app/javascript/src/constants/index.tsx:39]: yes, otherwise the nested routing won't work.

**Review by gowsik-ragunath — APPROVED** [review]
LGTM!

**Pull request merged**